app/src/main/java/org/simple/clinic/di/network/TimeoutCallFactory.kt

```diff
@@ -1,29 +1,41 @@
 package org.simple.clinic.di.network
 
-import okhttp3.Call
-import okhttp3.Request
-import retrofit2.Invocation
+import retrofit2.Call
+import retrofit2.CallAdapter
+import retrofit2.Retrofit
+import java.lang.reflect.Type
 
-class TimeoutCallFactory(val delegate: Call.Factory) : Call.Factory {
+class TimeoutCallFactory : CallAdapter.Factory() {
 
   companion object {
-    fun create(
-        delegate: Call.Factory
-    ): TimeoutCallFactory {
+    fun create(): TimeoutCallFactory {
       return TimeoutCallFactory(
-          delegate = delegate
       )
     }
   }
 
-  override fun newCall(request: Request): Call {
-    val call = delegate.newCall(request)
-    val tag = request.tag(Invocation::class.java)
-    val methodCalled = tag?.method()
-    val timeoutValueAndUnit = methodCalled?.getAnnotation(Timeout::class.java)
-    if (timeoutValueAndUnit != null) {
-      call.timeout().timeout(timeoutValueAndUnit.value, timeoutValueAndUnit.unit)
+  override fun get(returnType: Type, annotations: Array<out Annotation>, retrofit: Retrofit): CallAdapter<*, *> {
+    var timeout: Timeout? = null
+    for (annotation in annotations) {
+      if (annotation.annotationClass == Timeout::class.java) {
+        timeout = annotation as Timeout
+        break
+      }
+    }
+
```

Comment:
We can return `null` if `Timeout` is `null`, since there is no need to create a new `CallAdapter`